doc/src/tutorial/matrices.rst

```diff
@@ -461,7 +461,7 @@ by injecting a custom zero test with warnings enabled.
     ...         result = None
     ...
     ...     # Warnings if evaluated into None
-    ...     if result == None:
+    ...     if not isinstance(result, bool):
```

Comment:
Why not `is result is None`?

Reply:
Idk. Is it legitimate to get something other than `bool` in this context? This way we are verifying both the presence of a value _as well as_ the value itself. Of course `... is None` is very flexible, and less restricting approach though.

Reply:
`x.is_zero` should return `True`, `False` or `None` (if `x` has attribute `is_zero`). I believe that `result is None` is more efficient than `not isinstance(result, bool)`.

Reply:
Somehow, efficiency does not apply to doc tests examples :thinking:. But thanks for the important clarification on return type, that is has to be `Optional[bool]`. Now i see why `is [not] None` would be the right choice.